src/EditorFeatures/Text/Extensions.TextBufferContainer.cs

```diff
@@ -37,8 +37,8 @@ private TextBufferContainer(ITextBuffer editorBuffer)
             /// <summary>
             /// A weak map of all Editor ITextBuffers and their associated SourceTextContainer
             /// </summary>
-            private static readonly ConditionalWeakTable<ITextBuffer, TextBufferContainer> s_textContainerMap = new ConditionalWeakTable<ITextBuffer, TextBufferContainer>();
-            private static readonly ConditionalWeakTable<ITextBuffer, TextBufferContainer>.CreateValueCallback s_createContainerCallback = CreateContainer;
+            private static readonly ConditionalWeakTable<ITextBuffer, WeakReference<TextBufferContainer>> s_textContainerMap = new();
+            private static readonly ConditionalWeakTable<ITextBuffer, WeakReference<TextBufferContainer>>.CreateValueCallback s_createContainerCallback = CreateContainer;
```

Comment:
:-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/ :-/

Reply:
my brain is too tired to have any sesne if this is ok.  seems massive risky.

Reply:
As far as risk, the one less risky change could be to tweak the code in VisualStudioActiveDocumentTracker, but that'd be a tactical fix only -- it'd be easy for somebody else to make a similar mistake down the road and re-trigger the leak. An external caller to the .AsTextContainer() API could also trigger the same leak too. Long term I think this is the right change, but we may have other tactical solutions if we want to push this into 16.11 or 17.0.

Reply:
Also realized this needs one extra check, so moving back to draft.